command/meta.go

```diff
@@ -17,26 +18,61 @@ const width = uint(78)
 
 // meta contains the meta-options and functionality for all CTS commands
 type meta struct {
-	UI cli.Ui
+	UI    cli.Ui
+	flags *flag.FlagSet
 
 	helpOptions []string
 	port        *int
+	addr        *string
+
+	tls tls
 }
 
-func (m *meta) defaultFlagSet(name string) *flag.FlagSet {
-	flags := flag.NewFlagSet(name, flag.ContinueOnError)
-	m.port = flags.Int("port", config.DefaultPort, "The port to use for the Consul Terraform Sync API server")
-	flags.SetOutput(ioutil.Discard)
+type tls struct {
+	caPath     *string
+	caFile     *string
+	clientCert *string
+	clientKey  *string
+	sslVerify  *bool
+}
+
+const (
+	// Command line flag names
+	flagPort     = "port"
+	flagHTTPAddr = "http-addr"
 
-	flags.VisitAll(func(f *flag.Flag) {
+	flagCAPath     = "ca-path"
+	flagCaFile     = "ca-cert"
+	flagClientCert = "client-cert"
+	flagClientKey  = "client-key"
+	flagSSLVerify  = "ssl-verify"
+)
+
+func (m *meta) defaultFlagSet(name string) *flag.FlagSet {
+	m.flags = flag.NewFlagSet(name, flag.ContinueOnError)
+
+	// Values provide both default values, and documentation for the default value when -help is used
+	m.port = m.flags.Int(flagPort, config.DefaultPort,
+		fmt.Sprintf("The port to use for the Consul Terraform Sync API server, it is preferred to use the %s field instead", flagHTTPAddr))
+	m.addr = m.flags.String(flagHTTPAddr, api.DefaultAddress, "The address to use for the Consul Terraform Sync API server")
```

Comment:
Which flag would someone use with an address with the https scheme?

Reply:
They just need to include "https" in the address to enable the https scheme, this is what Consul does I believe, I pulled the scheme parsing code from there. So for example: https://localhost:8558 would set the https scheme, we probably would want to recommend a port for https as well, but that's more a server config thing.

Reply:
https://github.com/hashicorp/consul-terraform-sync/blob/4f234c37222bb86da9b165a9db19dff141500bee/api/client.go#L394

Reply:
Yeah my concern is the clarity in the flag name and description with its actual capabilities. Can you update the description to include HTTPS?

This is how Consul CLI has it documented
> The address and port of the Consul HTTP agent. The value can be
     an IP address or DNS address, but it must also include the port.
     This can also be specified via the CONSUL_HTTP_ADDR environment
     variable. The default value is http://127.0.0.1:8500. The scheme
     can also be set to HTTPS by setting the environment variable
     CONSUL_HTTP_SSL=true.

Reply:
ah I see, sure I'll add this

Reply:
@findkim added!

Reply:
Thank you!